Please dont hard code compiler paths

https://github.com/ptitSeb/box86/blob/21e9a43d211a6334fbb99652cfcc699b9e492915/CMakeLists.txt#L87-L89
Please leave that unset and force the user to use -DCMAKE_C_COMPILER if they are cross compiling/building :)
Currently you are breaking builds on systems that dont use the triple arm-linux-gnueabihf-gcc arch arm is one of them

cmake tells users how to set the compiler if it doesnt find it
```
CMake Error at CMakeLists.txt:92 (project):
  The CMAKE_C_COMPILER:

    arm-linux-gnueabihf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:92 (project):
  No CMAKE_ASM_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,9 +84,14 @@ string(COMPARE EQUAL "${CMAKE_HOST_SYSTEM_PROCESSOR}" "arm"  _arm)
 string(COMPARE EQUAL "${CMAKE_HOST_SYSTEM_PROCESSOR}" "armv8l"  _armv8l)
 
 if(_aarch64 AND (RK3399 OR ODROIDN2 OR TEGRAX1 OR TEGRA_T194 OR TEGRA_T234 OR RPI3ARM64 OR RPI4ARM64 OR PHYTIUM OR SD845 OR SD888 OR ADLINK OR A64 OR LX2160A OR RK3588 OR ARM64))
-    set(CMAKE_C_COMPILER arm-linux-gnueabihf-gcc)
-    #set(CMAKE_ASM_COMPILER arm-linux-gnueabihf-as) #will use gcc in fact
-    set(CMAKE_C_COMPILER_TARGET arm-linux-gnueabihf)
+    set(COMPILER_TRIPLE arm-linux-gnueabihf-gcc)
+    find_program(COMPILER_FOUND ${COMPILER_TRIPLE})
+    
+    if(COMPILER_FOUND)
+        set(CMAKE_C_COMPILER ${COMPILER_FOUND})
+    else()
+        message(STATUS "Cross compiler ${COMPILER_TRIPLE} not found, using CC or -DCMAKE_C_COMPILER")
+    endif()
 endif()
 
 project(box86 C ASM)

